textile/features.textile

```diff
@@ -1305,6 +1305,22 @@ h4. Message
 ** @(TM2e)@ @encoding@ string
 ** @(TM2i)@ @extras@ JSON-encodable object, used to contain any arbitrary key value pairs which may also contain other primitive JSON types, JSON-encodable objects or JSON-encodable arrays. The @extras@ field is provided to contain message metadata and/or ancillary payloads in support of specific functionality, e.g. push. Each of these supported extensions is documented separately; for 1.1 the only supported extension is @push@, via the @extras.push@ member; 1.2 adds the @delta@ extension whose keys and values are described by the attributes of the type @DeltaExtras@, and the @headers@ extension, which contains arbitrary @string->string@ key-value pairs, settable at publish time, and @ref@ whose keys and values are described by the attributes of the type @ReferenceExtras@. Unless otherwise specified, the client library should not attempt to do any filtering or validation of the @extras@ field itself, but should treat it opaquely, encoding it and passing it to realtime unaltered.
 ** @(TM2f)@ @timestamp@ time in milliseconds since epoch. If a message received from Ably does not contain a @timestamp@, it should be set to the @timestamp@ of the encapsulating @ProtocolMessage@
+** @(TM2j)@ @action@ enum - the @MessageAction@ type of the message. Any message received from Ably must have an @action@ field set to one of the following values:
+*** @(TM2j1)@ 0: @MESSAGE_UNSET@
+*** @(TM2j2)@ 1: @MESSAGE_CREATE@
+*** @(TM2j3)@ 2: @MESSAGE_UPDATE@
+*** @(TM2j4)@ 3: @MESSAGE_DELETE@
+*** @(TM2j5)@ 4: @MESSAGE_ANNOTATION_CREATE@
+*** @(TM2j6)@ 5: @MESSAGE_ANNOTATION_DELETE@
+** @(TM2K)@ @serial@ string - the serial of the message. This will be set to the @msgSerial@ of the @ProtocolMessage@ that encapsulates the message, with an additional offset that locates the message in the @ProtocolMessage@ messages array. If the message is received from Ably, this must be set.
```

Comment:
Should this be the "origin timeserial assigned to encapsulating `ProtocolMessage` after publish"?

Per below - the `msgSerial` field in the `ProtocolMessage` isn't a timeserial

Reply:
I've updated this, I still want to make it clear that it is the serial of the encapsulating ProtocolMessage, but with the Idx.